main_loop_tf/main.py

```diff
@@ -259,37 +259,38 @@ def __run(build_model):
     with tf.Graph().as_default() as graph:
         cfg.global_step = tf.Variable(0, trainable=False, name='global_step',
                                       dtype='int32')
-
-        # Create input placeholders for each gpu tower
-        # All the placeholders are instantiated with a default value
-        # since we want to be able to use a subset of them when the
-        # batchsize is not big enough to fill all th GPUs and TF raises
-        # a negative shape error if one placeholder is not fed (and has
-        # no default value).
-        # This is not a problem since when the default value is being
-        # used we ignore it's output and do not perform gradient descent
-        # on it.
-        # Note: shape for labels placeholders will be None since we
-        # provide flattened labels.
+        # Create a list of input placeholders for each GPU. We use
+        # placeholder_with_default because we use a subset of them when the
+        # batchsize is not big enough to fill all the GPUs and TF raises a
+        # negative shape error if one placeholder is not fed otherwise. This is
+        # not a problem, since when the default value is being used we ignore
+        # it's output and do not perform gradient descent on it.
         inputs_per_gpu = []
         val_inputs_per_gpu = []
         labels_per_gpu = []
-        for _ in range(cfg.num_splits):
+        for i, _ in enumerate(range(cfg.num_splits)):
             inputs_per_gpu.append(tf.placeholder_with_default(
-                np.zeros(shape=[1]+cfg.input_shape[1:], dtype=cfg._FLOATX),
-                shape=cfg.input_shape))
+                np.zeros(shape=[cfg.batch_size] + cfg.input_shape[1:],
```

Comment:
Is not better to use 1 as default value for the batch_size? In the case we do not feed the placeholder the tower results are not taken so I think it is not important which is the default size, so why do not preserve memory?